erts/lib_src/pthread/ethread.c

```diff
@@ -337,7 +341,11 @@ ethr_thr_create(ethr_tid *tid, void * (*func)(void *), void *arg,
     twd.stacksize = 0;
 
     if (opts && opts->name) {
+#ifdef __HAIKU__
+        snprintf(twd.name_buff, B_OS_NAME_LENGTH, "%s", opts->name);
```

Comment:
This differs from the declared size of `.name_buff[]`.  IMO this (and all other `snprintf`:s you change) should use `sizeof` of the destination array instead of hard-coding its size.

Reply:
Yes, I agree with @mikpe on the `sizeof` approach